Three/include/CGAL/Three/Scene_item.h

```diff
@@ -326,6 +326,18 @@ public Q_SLOTS:
   //!memory leak, for example when multiple items are created at the same time.
   virtual void itemAboutToBeDestroyed(Scene_item*);
 
+  //!When an item is erased, some of its internal data is deleted before
+  //!the actual item is. During this unstable state, signals can still be emited
+  //!and there is a risk of runtime crash. This boolean is used to specify if it is
+  //!safe to call an items internal data or not.
+  //! \see `itemAboutToBeDestroyed()`
+  virtual bool is_being_destroyed() const;
+
+  //!Use this when you override `itemAboutToBeDestroyed()` to avoid
+  //!a dangerous state.
+  //! \see `is_being_destroyed()`
+  virtual void set_being_destroyed(bool);
+
```

Comment:
I disagree with the addition of that pair of functions. The fact that items destruction is delayed should not be observable by any object of the application.